### PR TITLE
Add advice for event design.

### LINKED ISSUE #12: Events should be used for notification of actions, not actions themselves
@annevk sums this up [on public-webapps](http://lists.w3.org/Archives/Public/public-webapps/2014AprJun/0510.html), pointing out that dispatching a ClipboardEvent should not actually copy things to the user's clipboard.

He might add this advice to the DOM standard itself. But if he doesn't, then we should be sure to point it out at some point.

### PR DESCRIPTION
**Pull request by slightlyoff**
Fixes #12

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -338,21 +338,31 @@ So, if the object inherits from {{EventTarget}}, add a corresponding
 [=event handler IDL attributes=] on the {{GlobalEventHandlers}} interface, instead of directly on
 the relevant element interface(s).</p>
 
+<h2 id="event-design">Event Design</h2>
+
+<h3 id="events-are-for-notification">Events are for notification</h3>
+
+Try to design DOM events to deliver after-the-fact notifications of changes. It may be tempting to try to trigger side-effects from the action of <code>dispatchEvent</code>, but in general this is <a href="https://lists.w3.org/Archives/Public/public-webapps/2014AprJun/0510.html">discouraged</a>.
+
+<h3 id="favor-asynchronous-events">Favor asynchornous events</h3>
```
Comment (dbaron) [inline index.bs]: s/chorn/chron/

File: index.bs [hunk 2]
```diff
@@ -338,21 +338,31 @@ So, if the object inherits from {{EventTarget}}, add a corresponding
...
+
+A few events in the platform are specified to dispatch synchronously. These events cause problems for engines and performance issues in applications. If you feel you may ned a synchronous event in your design, please reconsider and ask the TAG for help in redesigning the API.
```
Comment (dbaron) [inline index.bs]: s/ned/need/

Or actually even "may ned" -> "need".


I also wonder if it's worth mentioning the pattern that IntersectionObserver uses here, although I'm not sure what to call it.

File: index.bs [hunk 3]
```diff
@@ -338,21 +338,31 @@ So, if the object inherits from {{EventTarget}}, add a corresponding
...
+
+<h3 id="state-and-subclassing">State and Event subclasses</h3>
+
+It's tempting to create subclasses of `Event` for all event types. This is frequently overkill. Consider subclassing `Event` when adding unique methods and large amounts of state. In all other cases, using a "vanilla" event with state captured in the `.target` object.
```
Comment (dbaron) [inline index.bs]: probably "overkill" -> "unnecessary".

File: index.bs (hunk 2 above)
Comment (travisleithead) [inline index.bs]: Do we want to go into the why these are problems? E.g., synchronous events cause re-entrancy (e.g., in a focus event handler another element is focused...) which can cause unexpected results in author code and are tricky to spec correctly. Also Sync mutation events were triggered at a time in which an operation of the DOM was partially-complete, resulting in reentrancy, breaking internal invariants, and causing platform reliability issues.

File: index.bs [hunk 4]
```diff
@@ -338,21 +338,31 @@ So, if the object inherits from {{EventTarget}}, add a corresponding
 [=event handler IDL attributes=] on the {{GlobalEventHandlers}} interface, instead of directly on
 the relevant element interface(s).</p>
```
Comment (travisleithead) [inline index.bs]: Wondering if the previous section should be relocated down into your new section...

File: index.bs [hunk 5]
```diff
@@ -338,21 +338,31 @@ So, if the object inherits from {{EventTarget}}, add a corresponding
...
 rare that its more specific ones are actually appropriate.
 
-Remember that JavaScript itself only has one numeric type, Number: IEEE 754 double-precision
-floating point, including ±0, ±Infinity, and NaN (although thankfully only one). The Web IDL
-"types" are simply coercion rules that apply when accepting an argument or triggering a setter. For
-example, a Web IDL <code>unsigned short</code> roughly says: "when someone passes this as an
-argument, take it modulo 65535 before doing any further processing". That is very rarely a useful
-thing to do.
+JavaScript has only one numeric type, Number: IEEE 754 double-precision
+floating point, including ±0, ±Infinity, and NaN (although thankfully only one). The Web IDL "types" are coercion rules that apply when accepting an argument or triggering a setter. For example, a Web IDL <code>unsigned short</code> roughly says: "when someone passes this as an argument, take it modulo 65535 before doing any further processing". That is very rarely a useful thing to do.
```
Comment (annevk) [inline index.bs]: Aren't the other NaNs exposed through typed array objects?

File: index.bs [hunk 6]
```diff
@@ -338,21 +338,31 @@ So, if the object inherits from {{EventTarget}}, add a corresponding
 [=event handler IDL attributes=] on the {{GlobalEventHandlers}} interface, instead of directly on
 the relevant element interface(s).</p>
 
+<h2 id="event-design">Event Design</h2>
+
+<h3 id="events-are-for-notification">Events are for notification</h3>
+
+Try to design DOM events to deliver after-the-fact notifications of changes. It may be tempting to try to trigger side-effects from the action of <code>dispatchEvent</code>, but in general this is <a href="https://lists.w3.org/Archives/Public/public-webapps/2014AprJun/0510.html">discouraged</a>.
```
Comment (annevk) [inline index.bs]: It's more than discouraged as it requires changing the DOM Standard each time it's done.

File: index.bs (hunk 1 above)
Comment (slightlyoff) [inline index.bs]: Done

File: index.bs (hunk 2 above)
Comment (slightlyoff) [inline index.bs]: Done

File: index.bs (hunk 3 above)
Comment (slightlyoff) [inline index.bs]: Done

File: index.bs (hunk 4 above)
Comment (slightlyoff) [inline index.bs]: Agree; moved

File: index.bs (hunk 2 above)
Comment (slightlyoff) [inline index.bs]: Added a note to this effect

File: index.bs (hunk 5 above)
Comment (slightlyoff) [inline index.bs]: Only by comparison operations, right?

File: index.bs (hunk 6 above)
Comment (slightlyoff) [inline index.bs]: Will update. Thanks.

**Pull request merged**